lua/modes.lua

```diff
@@ -192,7 +201,7 @@ M.disable_managed_ui = function()
 end
 
 M.setup = function(opts)
-	opts = opts or default_config
+	opts = vim.tbl_extend("keep", opts, default_config)
```

Comment:
This feels separate from the PR but I appreciate fixing our config merging. Should this be `vim.tbl_deep_extend` though?

Reply:
Yeah. Spot on. `tbl_extend` LGTM.